packages/calcite-components/src/components/dialog/dialog.tsx

```diff
@@ -298,12 +334,28 @@ export class Dialog
 
   @State() defaultMessages: DialogMessages;
 
+  @State() assistiveText = "";
+
+  @Watch("messages")
+  @Watch("dragEnabled")
+  @Watch("resizable")
+  updateAssistiveText(): void {
+    const { messages } = this;
+    this.assistiveText = messages
+      ? `${this.dragEnabled ? messages.dragEnabled : ""} ${this.resizable ? messages.resizeEnabled : ""}`.trim()
```

Comment:
Message concatenation should be avoided. Typically, we'd use a localized template ([example](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/components/input-time-zone/assets/input-time-zone/t9n/messages.json#L7)), but the conditional nature of this does make it tricky. @annierm18 Do you have any guidance here?

Reply:
If these are completely separate sentences, then this change should be fine. We have some guidance on string concatenation on our website here: https://esriis.sharepoint.com/sites/i18n_t9n/SitePages/String-Composition.aspx in the "Concatenation" section. I checked with Jonathan Turpin from the Translation team on this and he also said it should be fine. The major concatenation issue happens when you use concatenation within the same sentence.

Reply:
Makes sense. 👍